src/Decoder.c

```diff
@@ -4899,23 +4899,15 @@ static ZyanStatus ZydisDecodeInstruction(ZydisDecoderState* state,
 ZyanStatus ZydisDecoderInit(ZydisDecoder* decoder, ZydisMachineMode machine_mode,
     ZydisStackWidth stack_width)
 {
-    static const ZyanBool decoder_modes[ZYDIS_DECODER_MODE_MAX_VALUE + 1] =
-    {
+    static const ZyanU32 decoder_modes =
 #ifdef ZYDIS_MINIMAL_MODE
-        ZYAN_TRUE , // ZYDIS_DECODER_MODE_MINIMAL
-#else
-        ZYAN_FALSE, // ZYDIS_DECODER_MODE_MINIMAL
+        (1 << ZYDIS_DECODER_MODE_MINIMAL) ||
 #endif
-        ZYAN_FALSE, // ZYDIS_DECODER_MODE_AMD_BRANCHES
-        ZYAN_FALSE, // ZYDIS_DECODER_MODE_KNC
-        ZYAN_TRUE , // ZYDIS_DECODER_MODE_MPX
-        ZYAN_TRUE , // ZYDIS_DECODER_MODE_CET
-        ZYAN_TRUE , // ZYDIS_DECODER_MODE_LZCNT
-        ZYAN_TRUE , // ZYDIS_DECODER_MODE_TZCNT
-        ZYAN_FALSE, // ZYDIS_DECODER_MODE_WBNOINVD
-        ZYAN_TRUE,  // ZYDIS_DECODER_MODE_CLDEMOTE
-        ZYAN_FALSE  // ZYDIS_DECODER_MODE_IPREFETCH
-    };
+        (1 << ZYDIS_DECODER_MODE_MPX) ||
```

Comment:
I think these should use `|` instead of `||`

Reply:
Oops.